src/rt/sched/backpressure.h

```diff
@@ -0,0 +1,249 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+/**
+ * ## Introduction
+ *
+ * The primary goal of backpressure is to prevent runaway growth of a cown's
+ * message queue that might eventually result in a Verona application running
+ * out of memory. Cowns that receive messages at a higher rate than they are
+ * able to process are considered "overloaded" and may lead to this unbounded
+ * queue growth. Once an overloaded cown is identified, cowns sending messages
```

Comment:
This crosses lines so I'm not going to do a suggested edit.

How about...

"If a cown receives messages more quickly than it is able to process them, the queue will start to grow in an unbounded fashion. Cowns that aren't able to keep up with their arriving messages are marked as "overloaded". Once a cown is marked as overloaded, cowns sending messages to it will be temporarily descheduled ("muted"). Descheduling cowns that are sending to an overloaded cown helps alleviate pressure on the it by:

- Turning off a source of incoming messges
- Allows for more scheduler time to go to the overload cown and not any of the descheduled ones.

Similarly, cowns sending to cowns that have been muted will also be muted to prevent runaway queue growth while the muted cown in descheduled.

Muted cowns will be rescheduled after the receiver that triggered their muting (the "mutor") is no longer muted or overloaded.

Reply:
Updated.